conans/test/functional/graph_lock/graph_lock_ci_test.py

```diff
@@ -704,3 +704,61 @@ def test_version_ranges(self):
         self.assertIn("br/0.2", client.out)
         self.assertNotIn("br/0.1", client.out)
+
+
+class CIPrivateRequiresTest(unittest.TestCase):
+    def test(self):
+        # https://github.com/conan-io/conan/issues/7985
+        client = TestClient()
+        files = {
+            "private/conanfile.py": GenConanfile().with_option("myoption", [True, False]),
+            "pkga/conanfile.py": textwrap.dedent("""
+                from conans import ConanFile
+                class PkgA(ConanFile):
+                    requires = ("private/0.1", "private"),
+                    def configure(self):
+                        self.options["private"].myoption = True
+                """),
+            "pkgb/conanfile.py": textwrap.dedent("""
+                from conans import ConanFile
+                class PkgB(ConanFile):
+                    requires = "pkga/0.1", ("private/0.1", "private"),
+                    def configure(self):
+                        self.options["private"].myoption = False
+                """),
+            "pkgc/conanfile.py": GenConanfile().with_require("pkgb/0.1")
+        }
+        client.save(files)
+        client.run("export private private/0.1@")
+        client.run("export pkga pkga/0.1@")
+        client.run("export pkgb pkgb/0.1@")
+
+        client.run("lock create pkgc/conanfile.py --name=pkgc --version=0.1 --build "
+                   "--lockfile-out=conan.lock")
+        client.run("lock build-order conan.lock --json=build_order.json")
+        json_file = client.load("build_order.json")
+        to_build = json.loads(json_file)
+        if client.cache.config.revisions_enabled:
+            build_order = [[['private/0.1@',
+                             'd2560ba1787c188a1d7fabeb5f8e012ac53301bb', 'host', '3'],
+                            ['private/0.1@',
+                             '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9', 'host', '4']],
+                           [['pkga/0.1@',
+                             '5b0fc4382d9c849ae3ef02a57b62b26ad5137990', 'host', '2']],
+                           [['pkgb/0.1@',
+                             'd7d6ac48b43e368b0a5ff79015acea49b758ffdf', 'host', '1']]]
+        else:
+            build_order = [[['private/0.1@',
+                             'd2560ba1787c188a1d7fabeb5f8e012ac53301bb', 'host', '3'],
+                            ['private/0.1@',
+                             '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9', 'host', '4']],
+                           [['pkga/0.1@',
+                             '5b0fc4382d9c849ae3ef02a57b62b26ad5137990', 'host', '2']],
+                           [['pkgb/0.1@',
+                             'd7d6ac48b43e368b0a5ff79015acea49b758ffdf', 'host', '1']]]
+
+        self.assertEqual(to_build, build_order)
+
+        for ref, _, _, node_id in build_order[0]:
+            client.run("install %s --build=%s --lockfile=conan.lock --lockfile-out=conan.lock "
+                       "--lockfile-id=%s" % (ref, ref, node_id))
```

Comment:
Maybe check it's been built?

```suggestion
        for ref, pid, _, node_id in build_order[0]:
            client.run("install %s --build=%s --lockfile=conan.lock --lockfile-out=conan.lock "
                       "--lockfile-id=%s" % (ref, ref, node_id))
            self.assertIn('private/0.1:{} - Build'.format(pid))
```